src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DynamicDocumentContainer.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Razor.Workspaces
+{
+    internal abstract class DynamicDocumentContainer
+    {
+        public abstract string FilePath { get; }
+
+        public abstract TextLoader GetTextLoader(string filePath);
+
+        public abstract object GetMappingService();
```

Comment:
So we're trying to move off of Roslyn's `ISpanMappingService` and `IDocumentExcerptService` types (they're internal and only available to us because they IVT us) which is why i'm returning `object` here. I have changes flowing through Roslyn now that shims those types and creates new `IRazorSpanMappingService` and `IRazorDocumentExcerptService`  types that will be in a `Microsoft.CodeAnalysis.ExternalAccess.Razor` assembly (once we consume the updated packages). At that time these `object` returns iwll become `IRazorSpanMappingService` and `IRazorDocumentExcerptService`.

It doesn't feel right adding a comment about this currently because it's all currently in flux 😢. That being said, I definitely should have added the above blob to the description of this PR